components/cdc/src/endpoint.rs

```diff
@@ -378,24 +406,40 @@ impl<T: CasualRouter<RocksEngine>> Endpoint<T> {
         }
     }
 
-    fn on_region_ready(&mut self, region_id: u64, resolver: Resolver, region: Region) {
+    fn on_region_ready(&mut self, observe_id: ObserveID, resolver: Resolver, region: Region) {
+        let region_id = region.get_id();
         if let Some(delegate) = self.capture_regions.get_mut(&region_id) {
-            if let Err(e) = delegate.on_region_ready(resolver, region) {
-                assert!(delegate.has_failed());
-                // Delegate has error, deregister the corresponding region.
-                let deregister = Deregister::Region { region_id, err: e };
-                if let Err(e) = self.scheduler.schedule(Task::Deregister(deregister)) {
-                    error!("schedule cdc task failed"; "error" => ?e);
+            if delegate.id == observe_id {
+                if let Err(e) = delegate.on_region_ready(resolver, region) {
+                    assert!(delegate.has_failed());
+                    // Delegate has error, deregister the corresponding region.
+                    let deregister = Deregister::Region { region_id, err: e };
+                    if let Err(e) = self.scheduler.schedule(Task::Deregister(deregister)) {
+                        error!("schedule cdc task failed"; "error" => ?e);
+                    }
                 }
+            } else {
+                debug!("stale region ready"; "region_id" => region.get_id(), "observe_id" => ?observe_id, "current_id" => ?delegate.id);
             }
         } else {
-            warn!("region not found on region ready (finish building resolver)"; "region_id" => region_id);
+            debug!("region not found on region ready (finish building resolver)"; "region_id" => region.get_id());
         }
     }
 
     fn on_min_ts(&mut self, min_ts: TimeStamp) {
-        for delegate in self.capture_regions.values_mut() {
-            delegate.on_min_ts(min_ts);
+        let mut min_resolved_ts = TimeStamp::max();
+        let mut min_ts_region_id = 0;
+        for (region_id, delegate) in self.capture_regions.iter_mut() {
+            if let Some(resolved_ts) = delegate.on_min_ts(min_ts) {
+                if resolved_ts < min_resolved_ts {
+                    min_resolved_ts = resolved_ts;
+                    min_ts_region_id = *region_id;
+                }
+            }
+        }
+        if min_ts_region_id > 0 {
+            CDC_MIN_TS_REGION.set(min_ts_region_id as i64);
+            CDC_MIN_RESOLVED_TS.set((min_resolved_ts.physical() / 1000) as i64);
```

Comment:
As u64, it loses precision.

Reply:
It should be fine from `u64` to `i64`.

Reply:
I mean `999u64/1000u64 = 0u64` instead of 0.999.

Reply:
Oh, got it